src/main/java/org/candlepin/model/activationkeys/ActivationKeyCurator.java

```diff
@@ -95,7 +96,10 @@ public List<ActivationKey> findByKeyNames(String ownerKey, Collection<String> ke
                 .setParameter("keys", keyNameBlock)
                 .getResultList());
         }
-
+        List<String> sortOrder = new ArrayList<>();
+        sortOrder.addAll(keyNames);
+        Collections.sort(foundActivationKeys,
+            Comparator.comparing(item -> sortOrder.indexOf(item.getName())));
```

Comment:
While this works, we should not be doing this globally at the curator level. The `ConsumerResource.findActivationKeys` method is a prime spot to do this, as that's the only time we'll care about ensuring the input and output retain the same ordering. It also saves us from having to create a new copy of the list since we'll know the exact type and ordering already.

As a bonus, we also move the testing of the ordering to the resource level where it belongs as it's become part of the registration spec now.

Reply:
To add to what @Ceiu said, can we rename and/or add a javadoc on the findActivationKeys method so that developers know that it finds activation keys while keeping the order? Maybe `findActivationKeysInOrder`?

Reply:
Still needs a spec test. I pushed it prematurely.